app/src/main/java/com/example/musicapp/data/repository/SongRepository.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.musicapp.data.repository
+
+import android.util.Log
+import com.example.musicapp.data.model.Song
+import com.example.musicapp.data.source.SongDataSource
+import com.example.musicapp.data.source.local.SongLocalDataSource
+import com.example.musicapp.utils.OnDataLoadCallBack
+
+class SongRepository private constructor(val localSource: SongLocalDataSource) : SongDataSource {
```

Comment:
Giao tiếp = interface
```suggestion
class SongRepository private constructor(val localSource: SongDataSource) : SongDataSource {
```